File: backend/internal/domain/project.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Project struct {
	Id          uuid.UUID `json:"id"`
	UserId      uuid.UUID `json:"user_id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`

	Members []ProjectMember `json:"members,omitempty"`
	Columns []ProjectColumn `json:"columns,omitempty"`
}

func (p *Project) IsMember(userId uuid.UUID) bool {
	for _, member := range p.Members {
		if member.UserId == userId {
			return true
		}
	}
	return false
}

func (p *Project) IsCreator(userId uuid.UUID) bool {
	for _, member := range p.Members {
		if member.UserId == userId && member.Role == ProjectMemberRoleCreator {
			return true
		}
	}
	return false
}

type ProjectMemberRole string

var (
	ProjectMemberRoleCreator ProjectMemberRole = "creator"
	ProjectMemberRoleMember  ProjectMemberRole = "member"
)

type ProjectMember struct {
	Id        uuid.UUID         `json:"id"`
	UserId    uuid.UUID         `json:"user_id"`
	User      *User             `json:"user,omitempty"`
	ProjectId uuid.UUID         `json:"project_id"`
	Role      ProjectMemberRole `json:"role"`
}

type ProjectColumn struct {
	Id           uuid.UUID `json:"id"`
	ProjectId    uuid.UUID `json:"project_id"`
	Name         string    `json:"name"`
	Description  string    `json:"description"`
	Color        string    `json:"color"`
	Position     int       `json:"position"`
	IsDoneColumn bool      `json:"is_done_column"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

File: frontend/src/types/project.ts
import type { User } from './user';

export interface Project {
  id: string;
  user_id: string;
  name: string;
  description: string;
  repository_url: string;
  repository_owner: string;
  repository_name: string;
  default_branch: string;
  branch_name_prefix: string;
  created_at: string;
  updated_at: string;
  members: Member[];
  columns: ProjectColumn[];
}

export interface ProjectColumn {
  id: string;
  project_id?: string;
  name: string;
  description: string;
  color: string;
  position: number;
  is_done_column: boolean;
  created_at?: string;
  updated_at?: string;
}

export interface Member {
  id: string;
  user_id: string;
  user: User;
  project_id: string;
  role: string;
}

export enum ProjectMemberRole {
  Creator = 'creator',
  Member = 'member',
}
